odb-compiler/include/odbc/util/TGCPlugin.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "odbc/config.hpp"
+#include <string>
+
+namespace odbc {
+
+class KeywordDB;
+
+class TGCPlugin
+{
+public:
+    TGCPlugin() = delete;
+    TGCPlugin(const TGCPlugin& other) = delete;
+    TGCPlugin(TGCPlugin&& other) = default;
+    ~TGCPlugin();
+
+    static TGCPlugin* load(const char* filename);
```

Comment:
Tbh we should really use RAII with `std::unique_ptr` or `std::optional` here IMO

Reply:
```std::unique_ptr``` would definitely be better